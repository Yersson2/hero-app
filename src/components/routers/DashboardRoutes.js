import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { DcPage } from '../dc/DcPage'
import { HeroPage } from '../heroes/HeroPage'
import { MarvelPage } from '../marvel/MarvelPage'
import { SearchPage } from '../search/SearchPage'
import { Navbar } from '../ui/Navbar'

export const DashboardRoutes = () => {
  return (
    <>
      <Navbar />
      <div className = "container mt-2">
        <Switch>
          <Route exact path = "/marvel" component = { MarvelPage } />
          <Route exact path = "/hero/:heroId" component = { HeroPage } />
          <Route exact path = "/dc" component = { DcPage } />
          <Route exact path = "/search" component = { SearchPage } />

          <Redirect to = "/marvel" />
        </Switch>
      </div>
    </>
  )
}
