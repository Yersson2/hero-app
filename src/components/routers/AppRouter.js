import React, { useContext } from 'react';
import {
  BrowserRouter as Router,
  Switch
} from "react-router-dom";
import { AuthContext } from '../../auth/AuthContext';
import { LoginPage } from '../login/LoginPage';
import { PublicRoute } from './PublicRoute';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';

export const AppRouter = () => {

  const { user } = useContext( AuthContext );

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute exact path = "/login" component = { LoginPage } isAuthenticated = { user.logged } />
          <PrivateRoute path = "/" component = { DashboardRoutes } isAuthenticated = { user.logged } />
        </Switch>
      </div>
    </Router>
  )
}
