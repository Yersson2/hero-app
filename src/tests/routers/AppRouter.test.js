import { mount } from 'enzyme';
import React from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from '../../components/routers/AppRouter';

describe('test un AppRouter', () => {

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false
    }
  };

  test('should show the login if does not authenticated', () => {

    const wrapper = mount(
      <AuthContext.Provider value = { contextValue } >
        <AppRouter />
      </AuthContext.Provider>
    );

    expect( wrapper ).toMatchSnapshot();
    
  });

  test('should show the marvel component if it is authenticated', () => {

    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged: true,
        name: 'Yersson'
      }
    };

    const wrapper = mount(
      <AuthContext.Provider value = { contextValue } >
        <AppRouter />
      </AuthContext.Provider>
    );

    expect( wrapper.find('.navbar').exists() ).toBe( true ); 
    
  })
  
  
  
})
