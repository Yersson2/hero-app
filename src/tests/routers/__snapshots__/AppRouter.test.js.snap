// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test un AppRouter should show the login if does not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login Page
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
